### ISSUE TITLE
deal with NaN and write tests for crop_suitability

### ISSUE THREAD

**Issue by remicousin** [iridl/python-maprooms#365]
I don't see any reason you couldn't write tests for functions in maproom.py. Do you have a specific impediment in mind? Writing tests for the UI requires infrastructure we don't have in place yet, but a function that's pure computation like this one should be testable using the same paradigm you've been using elsewhere.

I'm ok deferring proper handling of NaNs if you want.

_Originally posted by @aaron-kaplan in https://github.com/iridl/python-maprooms/pull/280#discussion_r1268604277_

**Comment by remicousin** [comment]
@aaron-kaplan , would it not be simpler to take functions that are not UI but pure calculation outside of maproom.py for testing purposes? See what I get when getting started naively





(enactsmulti) remic@MacBook-Pro-RC-3:~/Dash/python-maprooms/enacts (remic/crop_suit)$ python -m pytest tests/test_crop_suitaiblity_mr.py
============================================================= test session starts =============================================================
platform darwin -- Python 3.9.13, pytest-7.2.0, pluggy-1.0.0
rootdir: /Users/remic/Dash/python-maprooms/enacts
plugins: dash-2.7.0
collected 0 items / 1 error                                                                                                                   

=================================================================== ERRORS ====================================================================
_____________________________________________ ERROR collecting tests/test_crop_suitaiblity_mr.py ______________________________________________
tests/test_crop_suitaiblity_mr.py:1: in <module>
    from crop_suitability import maproom_crop_suit as mcs
crop_suitability/__init__.py:1: in <module>
    from . import maproom_crop_suit
crop_suitability/maproom_crop_suit.py:13: in <module>
    from . import layout_crop_suit
crop_suitability/layout_crop_suit.py:16: in <module>
    from globals_ import GLOBAL_CONFIG
globals_.py:7: in <module>
    GLOBAL_CONFIG = pingrid.load_config("config-defaults.yaml:" + os.environ["CONFIG"])
../../../opt/miniconda3/envs/enactsmulti/lib/python3.9/os.py:679: in __getitem__
    raise KeyError(key) from None
E   KeyError: 'CONFIG'
=========================================================== short test summary info ===========================================================
ERROR tests/test_crop_suitaiblity_mr.py - KeyError: 'CONFIG'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by aaron-kaplan** [comment]
Yes, you're right. Ideally files wouldn't do stuff like this at import time so we could test any file, but until we resolve that problem, put non-dash functions in a separate file for testability.

**Comment by remicousin** [comment]
ok. Can we wrap up PR #351  first then? I had rearranged the tests there and so I would put crop_suitability function in calc and have the tests as set in that PR, unless it's not a good idea.
